applications/nrf5340_audio/src/bluetooth/ble_core.c

```diff
@@ -267,3 +273,162 @@ int ble_core_init(ble_core_ready_t ready_callback)
 
 	return 0;
 }
+
+static void shell_help_print(const struct shell *shell)
+{
+	shell_print(shell, "Possible TX power values:");
+	shell_print(shell, "\tzero: 0dBm");
+	shell_print(shell, "\tneg1: -1dBm");
+	shell_print(shell, "\tneg2: -2dBm");
+	shell_print(shell, "\tneg3: -3dBm");
+	shell_print(shell, "\tneg4: -4dBm");
+	shell_print(shell, "\tneg5: -5dBm");
+	shell_print(shell, "\tneg6: -6dBm");
+	shell_print(shell, "\tneg7: -7dBm");
+	shell_print(shell, "\tneg8: -8dBm");
+	shell_print(shell, "\tneg12: -12dBm");
+	shell_print(shell, "\tneg16: -16dBm");
+	shell_print(shell, "\tneg20: -20dBm");
+	shell_print(shell, "\tneg40: -40dBm");
+}
+
+static enum ble_hci_vs_tx_power tx_power_input_check_and_enum_return(const char *input)
+{
+	if (strcmp(input, "zero") == 0) {
+		return BLE_HCI_VSC_TX_PWR_0dBm;
+	} else if (strcmp(input, "neg1") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg1dBm;
+	} else if (strcmp(input, "neg2") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg2dBm;
+	} else if (strcmp(input, "neg3") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg3dBm;
+	} else if (strcmp(input, "neg4") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg4dBm;
+	} else if (strcmp(input, "neg5") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg5dBm;
+	} else if (strcmp(input, "neg6") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg6dBm;
+	} else if (strcmp(input, "neg7") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg7dBm;
+	} else if (strcmp(input, "neg8") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg8dBm;
+	} else if (strcmp(input, "neg12") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg12dBm;
+	} else if (strcmp(input, "neg16") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg16dBm;
+	} else if (strcmp(input, "neg20") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg20dBm;
+	} else if (strcmp(input, "neg40") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg40dBm;
+	} else {
+		return BLE_HCI_VSC_TX_PWR_INVALID;
+	}
+}
+
+static void conn_tx_pwr_set(struct bt_conn *conn, void *data)
```

Comment:
Prefer if this function returned error codes.

Reply:
Hmm it does not look like there is an easy way of doing this when using bt_conn_foreach()